events/iot.go

```diff
@@ -0,0 +1,34 @@
+package events
+
+// IoTCustomAuthorizerRequest contains data coming in to a custom IoT device gateway authorizer function.
+type IoTCustomAuthorizerRequest struct {
+	HttpContext        *IoTHttpContext `json:"httpContext,omitempty"`
+	MqttContext        *IoTMqttContext `json:"mqttContext,omitempty"`
+	TlsContext         *IotTlsContext  `json:"tlsContext,omitempty"`
+	AuthorizationToken string          `json:"token"`
+	TokenSignature     string          `json:"tokenSignature"`
+}
+
+type IoTHttpContext struct {
+	Headers     map[string]string `json:"headers,omitempty"`
+	QueryString string            `json:"queryString"`
+}
+
+type IoTMqttContext struct {
+	ClientId string `json:"clientId"`
```

Comment:
style: general field and  type naming conventions should consistently capitalize these initialisms 

```
Id -> ID
HTTP -> HTTP
Tls->TLS
Mqtt -> MQTT
```